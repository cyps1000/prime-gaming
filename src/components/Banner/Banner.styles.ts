/**
 *  Material UI Imports
 * @see https://material-ui.com/styles/basics/
 */
import { makeStyles } from "@material-ui/core/styles";

/**
 * Styles the component
 */
const useStyles = makeStyles((theme) => ({
  root: {},
  img: {
    height: "42rem",
    width: "118.96rem",
    backgroundPosition: "center",
    backgroundRepeat: "no-repeat",
    backgroundSize: "cover",
    position: "relative",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    background: "no-repeat center center fixed",
    [theme.breakpoints.down("sm")]: {
      height: "inherit",
      width: "-webkit-fill-available",
    },
    [theme.breakpoints.down("md")]: {
      width: "-webkit-fill-available",
    },
  },
}));

export { useStyles };
