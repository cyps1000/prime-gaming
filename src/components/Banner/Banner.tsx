/**
 * Imports i18n
 */
import { useTranslation } from "react-i18next";

/**
 * Material UI Imports
 */
import Grid from "@material-ui/core/Grid";

/**
 * Assets Imports
 */
import BannerImage from "./banner.png";

/**
 * Imports the component styles
 */
import { useStyles } from "./Banner.styles";

/**
 * Defines the props interface
 */
export interface BannerProps {
  text?: string;
}

/**
 * Displays the component
 */
const Banner: React.FC<BannerProps> = (props) => {
  const { text } = props;

  /**
   * Handles the translations
   */
  const { t } = useTranslation();

  /**
   * Gets the component styles
   */
  const classes = useStyles({ banner: BannerImage });

  return (
    <div className={classes.root}>
      {/* <Grid container spacing={0}>
        <Grid item xs={12} sm={12} md={12}>
          <img className={classes.img} src={BannerImage} alt="banner" />
        </Grid>
      </Grid> */}
    </div>
  );
};

export default Banner;
