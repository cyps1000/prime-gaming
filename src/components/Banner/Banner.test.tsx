/**
 * @see https://testing-library.com/docs/react-testing-library/intro
 * @see https://www.robinwieruch.de/react-testing-library
 */
import { render } from "@testing-library/react";

/**
 * External Imports
 */
import pretty from "pretty";

/**
 * Imports component
 */
import Banner from "./Banner";

/**
 * Default test
 */
describe("Banner Rendering Tests", () => {
  it("Renders the component without errors", () => {
    render(<Banner />);
  });
});

/**
 * Banner snapshot test
 */
describe("Banner Snapshot Test", () => {
  it("Passes the snapshot test", () => {
    const { container } = render(<Banner />);

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
