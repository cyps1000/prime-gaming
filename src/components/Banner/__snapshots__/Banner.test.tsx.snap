// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Snapshot Test Passes the snapshot test 1`] = `"<div class=\\"makeStyles-root-4 makeStyles-root-6\\"></div>"`;
